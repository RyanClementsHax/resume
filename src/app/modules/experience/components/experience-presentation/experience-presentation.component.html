<article class="container">
  <img [src]="experience.imgUrl" />
  <div class="details">
    <header class="details-header">
      <h2>{{ experience.title }}</h2>
      <em class="subtitle">
        <h3>{{ experience.position }},&nbsp;{{ experience.company }}</h3>
        <span>
          <time>{{ experience.startDate | date:'MMM/yyyy' }}</time>
          &nbsp;-&nbsp;
          <time>{{ experience.endDate ? (experience.endDate | date:'MMM/yyyy') : 'Present' }}</time>
        </span>
      </em>
    </header>
    <p>{{ experience.description }}</p>
    <footer class="details-footer">
      <div class="linkUrls">
        <span *ngFor="let link of experience.linkUrls" [title]="link.disabledTitle"
          [ngClass]="link.disabledTitle ? 'disabled-wrapper' : ''">
          <a [href]="link.url" [ngClass]="link.disabledTitle ? 'disabled' : ''" class="linkUrl"
            rel="noopener noreferrer" target="_blank">
            {{ link.text }}
          </a>
        </span>
      </div>
      <div class="technologies">
        <span class="tech" *ngFor="let tech of experience.technologies">
          {{ tech }}
        </span>
      </div>
    </footer>
  </div>
</article>
