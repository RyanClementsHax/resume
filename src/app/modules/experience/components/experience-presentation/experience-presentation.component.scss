@import "variables.scss";

:host {
    display: block;
    
    .container {
        display: flex;
        flex-direction: column;
        align-items: center;
        max-width: $content-width;
        padding: $content-padding;
        border: $light-border;
        border-radius: $default-border-radius;
        box-shadow: $default-box-shadow;

        img {
            width: 100%;
            max-width: 100%;
            border-radius: $default-border-radius;
        }

        .details {
            display: flex;
            flex-direction: column;
            height: auto;
            width: 100%;

            h2 {
                margin: 0;
            }

            .details-footer {
                margin-top: auto;

                .tech {
                    display: inline-block;
                    padding: 0 5px;
                    margin-top: 3px;
                    margin-right: 3px;
                    border-radius: $default-border-radius;
                    background: $color-background-grey;
                    color: $color-accent;
                }
            }
        }

        @media screen and (min-width: $small-screen-break-point) {
            flex-direction: row;
            align-items: flex-start;

            img {
                width: 30%;
            }

            .details {
                margin-left: $content-padding;
                align-self: stretch;
            }
        }
    }
}