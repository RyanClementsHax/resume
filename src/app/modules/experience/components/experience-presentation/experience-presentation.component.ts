import { Component, Input } from '@angular/core';
import { Experience } from '../../services/experience.service';

@Component({
  selector: 'experience-presentation',
  templateUrl: './experience-presentation.component.html',
  styleUrls: ['./experience-presentation.component.scss']
})
export class ExperiencePresentationComponent {
  @Input() experience: Experience = {
    title: '',
    company: '',
    position: '',
    startDate: new Date(0),
    description: '',
    technologies: [],
    imgUrl: ''
  };
}
