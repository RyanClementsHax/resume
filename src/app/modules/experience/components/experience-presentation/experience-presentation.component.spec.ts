import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { NO_ERRORS_SCHEMA } from '@angular/core';

import { ExperiencePresentationComponent } from './experience-presentation.component';

describe('ExperiencePresentationComponent', () => {
  let component: ExperiencePresentationComponent;
  let fixture: ComponentFixture<ExperiencePresentationComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [ExperiencePresentationComponent],
      schemas: [NO_ERRORS_SCHEMA]
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(ExperiencePresentationComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });
});
