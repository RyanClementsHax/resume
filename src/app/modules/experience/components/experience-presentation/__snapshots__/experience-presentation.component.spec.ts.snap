// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExperiencePresentationComponent should match snapshot 1`] = `
ExperiencePresentationComponent {
  "experience": Object {
    "company": "",
    "description": "",
    "imgUrl": "",
    "position": "",
    "startDate": 1970-01-01T00:00:00.000Z,
    "technologies": Array [],
    "title": "",
  },
}
`;
